### PR TITLE
[WIP] txt diagnosis tool

### PR DESCRIPTION
**Pull request by emily101-gif**

### CODE REVIEW

**Comment by emily101-gif** [comment]
@pcm32 . Would you mind having a look at the python script for this tool. The tool works, and does produces a summary of the input.txt file (without errors). However the file with errors (NAN / character strings) does not show any warnings when running.

**Comment by pcm32** [comment]
This now passes the test, hopefully detecting what you were after @emily101-gif. Is the python code the original one from Immport? I would guess that there are more sophisticated ways of checking for NaNs in a pandas table that having to iterate on rows and columns, could you give that a quick look @emily101-gif?

**Comment by pcm32** [comment]
I would rather see something like this: https://stackoverflow.com/questions/14016247/find-integer-index-of-rows-with-nan-in-pandas-dataframe

File: flowtools/txt_diagnosis/txtdiagnosis.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    except ValueError:
+        return False
+
+
+def error_report(input_file, fname, output_file):
+    errors = 0
+    df = pd.read_table(input_file)
+    with open(output_file, "w") as outf:
+        for cols in df.columns.values:
+            if df[cols].count() != len(df[cols]):
```
Comment (pcm32) [inline flowtools/txt_diagnosis/txtdiagnosis.py:24]: Please give it a go at using the pandas dataframe more natively here for finding out in a couple of lines the offending lines with NaNs, and then reporting that in the same way.

**Review by pcm32 — CHANGES_REQUESTED** [review]
Well done! A few things to improve this PR, thanks!

**Review by pcm32 — APPROVED** [review]
We will leave this for now in the interest of time

**Pull request merged**